### PR TITLE
Add `maxTimeMS` to FindOptions

### PR DESCRIPTION
**Pull request by BrunoBernardino**
This is allowed, works, and is already being sent over, so it's really just a type thing.

### CODE REVIEW

**Comment by lucsoft** [comment]
can you add a test case please?

File: tests/cases/03_curd.ts [hunk 1]
```diff
@@ -578,3 +578,51 @@ testWithTestDBClient("testFindEmptyAsyncIteration", async (db) => {
 
   await db.collection("mongo_test_users").drop();
 });
+
+testWithTestDBClient("testFindWithMaxTimeMS", async (db) => {
```
Comment (BrunoBernardino) [inline tests/cases/03_curd.ts:582]: @lucsoft I hope this is enough!

File: tests/cases/03_curd.ts [hunk 2]
```diff
@@ -578,3 +578,51 @@ testWithTestDBClient("testFindEmptyAsyncIteration", async (db) => {
...
+  try {
+    await users.find({
+      uid: 0,
+      $where: "sleep(5) || true",
+    }, { maxTimeMS: 1 }).toArray();
+    assert(false);
+  } catch (e) {
+    assertEquals(Boolean(e.ok), false);
+    assertEquals(e.codeName, "MaxTimeMSExpired");
+    assertEquals(e.errmsg, "operation exceeded time limit");
```
Comment (BrunoBernardino) [inline tests/cases/03_curd.ts:612]: This is failing on 4.0 but I don't know what the error message actually is, there. Do you have any insight on how to differently check/assert depending on the mongo versions?

**Comment by lucsoft** [comment]
one more thing the spec says that maxTimeMS should be a function like find().maxTimeMS() not an option parameter

but the tests look good! thanks so much!

File: tests/cases/03_curd.ts (hunk 2 above)
Comment (lucsoft) [reply tests/cases/03_curd.ts:612]: im a bit again doing assert based on mongodb version as they should be compatible in that case

File: tests/cases/03_curd.ts (hunk 1 above)
Comment (lucsoft) [reply tests/cases/03_curd.ts:582]: Great!

**Comment by manyuanrong** [comment]
Before 4.4, the documentation didn't explain much about errObj, and the codeName field might not be available before 4.2. 
refs: 
https://www.mongodb.com/docs/v4.0/reference/method/db.getLastErrorObj/#db.getLastErrorObj
https://www.mongodb.com/docs/v4.4/reference/method/db.getLastErrorObj/#db.getLastErrorObj

**Comment by BrunoBernardino** [comment]
> one more thing the spec says that maxTimeMS should be a function like find().maxTimeMS() not an option parameter
> 
> but the tests look good! thanks so much!

I think I've seen that for the mongo shell function but not for the node.js driver documentation. Can you link to where you saw that?

**Comment by BrunoBernardino** [comment]
Took quite a while, but I finally nailed the problem. `findOne` doesn't seem to support `maxTimeMS` in mongo 4.0, only 4.2+, so it wasn't failing. I'm not certain if it's an implementation bug (unlikely since it works as expected for 4.2+), or simply something I didn't see in the documentation. You can see my journey in the commits, but hopefully the tests are good now.

**Pull request merged**